src/main/java/com/google/sps/servlets/PostQuestionServlet.java

```diff
@@ -0,0 +1,122 @@
+// Copyright 2019 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package com.google.sps.servlets;
+
+import com.google.sps.classes.Constants;
+import com.google.sps.classes.Utility;
+import java.io.IOException;
+import java.sql.Connection;
+import java.sql.DriverManager;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Timestamp;
+import java.util.Date;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.annotation.WebServlet;
+import javax.servlet.http.HttpServlet;
+import javax.servlet.http.HttpServletRequest;
+import javax.servlet.http.HttpServletResponse;
+
+/** 
+ * This servlet will post a question to the forum.
+ */
+@WebServlet("/post-question")
+public class PostQuestionServlet extends HttpServlet {
+
+  /** 
+   * This method will execute the query to post a question to the database.
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    String title = request.getParameter("question-title");
+    String body = request.getParameter("question-body");
+    int asker_id = Utility.getUserId();
+
+    // First we query the number of questions that exist so that we can update the
+    // QuestionFollower table as well.
+    try {
+      Connection connection = DriverManager
+          .getConnection(Utility.SQL_LOCAL_URL, Utility.SQL_LOCAL_USER, Utility.SQL_LOCAL_PASSWORD);
+      insertNewQuestion(connection, title, body, asker_id);
+      int newQuestionId = getNewQuestionId(connection);
+      insertNewFollower(connection, newQuestionId, asker_id);
```

Comment:
nit: you could do 'insertNewFollower(connection, getNewQuestionId(connection))', or even move the call to getNewQuestionId into insertNewFollower, since all you need is the Connection instance.

Reply:
done